### ISSUE TITLE
[JENKINS-23364] Rename hudson-dev:run to jenkins-dev:run

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#4696]
<p>To run Jenkins in development mode, you currently type:</p>

<p>    mvn hudson-dev:run</p>

<p>This command should probably be updated to:</p>

<p>    mvn jenkins-dev:run</p>

<p>See discussion here: <a href="https://groups.google.com/forum/#!topic/jenkinsci-dev/UNW6bWjcgF4" class="external-link" target="_blank" rel="nofollow noopener">https://groups.google.com/forum/#!topic/jenkinsci-dev/UNW6bWjcgF4</a></p>

---
<details><summary><i>Originally reported by <a title="Kevin Burke" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kevinburke">kevinburke</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-23364" target="_blank">Rename hudson-dev:run to jenkins-dev:run</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Harald Albers" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=albers">albers</a>
<li><b>reported by</b>: kevinburke
<li><b>status</b>: Resolved
<li><b>priority</b>: Minor
<li><b>resolution</b>: Won't Do
<li><b>resolved</b>: 2019-11-12T15:06:59+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=albers">albers</a>:</i>
<p>updated <a href="https://wiki.jenkins-ci.org/display/JENKINS/Building+Jenkins" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Building+Jenkins</a>: use jenkins-dev:run instead of hudson-dev:run</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=albers">albers</a>:</i>
<p>I was incorrect about the maven-hpi-plugin. Both <tt>hudson-dev:run</tt> and <tt>jenkins-dev:run</tt> are provided by maven-jenkins-dev-plugin.</p>

<p><tt>hudson-dev:run</tt> works out of the box because maven-jenkins-dev-plugin explicitly declares hudson-dev as the goalPrefix. For <tt>jenkins-dev:run</tt> to work, the plugin group declaration in settings.xml is necessary.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Harald Albers: Can this be resolved then? Seems to be configuration/documentation issue, and you resolved the latter.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=albers">albers</a>:</i>
<p>I'd like to make the <tt>jenkins-dev</tt> prefix available without further configuration, as suggested in the issue.</p>

<p>maven-jenkins-dev-plugin explicitely declares <tt>hudson-dev</tt> as  its <tt>goalPrefix</tt>.<br/>
My understanding of <a href="http://maven.apache.org/guides/introduction/introduction-to-plugin-prefix-mapping.html" class="external-link" target="_blank" rel="nofollow noopener">http://maven.apache.org/guides/introduction/introduction-to-plugin-prefix-mapping.html</a> is that this setting makes the default prefix <tt>jenkins-dev</tt> (derived from the <tt>artifactId</tt>), unavailable to default plugin resolution.<br/>
Thus, removing the <tt>goalPrefix</tt> declaration should make <tt>jenkins-dev</tt> available.</p>

<p>But this is my first contact with maven plugin programming, so any advice is welcome.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=albers">albers</a>:</i>
<p><tt>mvn jenkins-dev:run</tt> ist available only after applying special maven configuration while <tt>mvn hudson-dev</tt> works out of the box.</p>

<p>This behavior should be reversed.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=oleg_nenashev">oleg_nenashev</a>:</i>
<p>Reversing the behavior could be a breaking change.<br/>
Is it possible to a support both call options out-of-the-box (e.g. by parametrizing the prefix)?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=albers">albers</a>:</i>
<p>The proposed change only covers interactive invocation from the command line (<tt>mvn jenkins-dev:run</tt> vs <tt>mvn hudson-dev:run</tt>). It has no impact on how the goals are called from existing build scripts (configuring goal exectuion in certain phases in pom.xml).</p>

<p>As far as I know, there is no way of parameterising the prefix. There can only be one.</p>

<p>In order to support both prefixes, the plugin has to be renamed back to <em>maven-hudson-dev-plugin</em> and a <em>goalPrefix</em> of <em>jenkins-dev</em> has to be configured in the plugin. But this would impact existing builds.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick">jglick</a>:</i>
<p>Should be irrelevant after <a href="https://github.com/jenkinsci/jenkins/pull/4351" class="external-link" target="_blank" rel="nofollow noopener">jenkins #4351</a>.</p>
